test/page.spec.js

```diff
@@ -1653,6 +1654,15 @@ module.exports.addTests = function({testRunner, expect, headless}) {
       });
       expect(screenshot).toBeGolden('screenshot-grid-fullpage.png');
     });
+    it('should take fullPage screenshots when defaultViewport is null', async({server}) => {
+      const browser = await puppeteer.launch({ args: ['--no-sandbox'], defaultViewport: null });
```

Comment:
Would it be better to pass `defaultBrowserOptions` here:
https://github.com/GoogleChrome/puppeteer/blob/85aca8e1a56b5698f447ffd55b148482bf8db7ae/test/test.js#L157

and use it like:

    const browser = await puppeteer.launch(Object.assign({}, defaultBrowserOptions, { defaultViewport: null }));

?

Reply:
yep; you'll get defaultBrowesrOptions once you move the test to `puppeteer.spec.js`